src/precedence.js

```diff
@@ -2,61 +2,67 @@
-    var temporaryResult = this.numbers[0];
-    var finalResult = 0;
-    for (var i = 0; i < this.operations.length; i++) {
-        if (this.operations[i] === 'x') {
-            temporaryResult *= this.numbers[i + 1];
-        } else if (this.operations[i] === '/') {
-            temporaryResult /= this.numbers[i + 1];
-        } else {
-            var sign = this.operations[i] === '+' ? 1 : -1;
-            if (i !== this.operations.length - 1 && this.operations[i + 1] === '/' || this.operations[i + 1] === 'x') {
-                finalResult += temporaryResult;
-                temporaryResult = sign * this.numbers[i + 1];
+class Precedence {
+
+    /**
+     *  Constructor.  
+     */
+    constructor() {
+        this.numbers = [];
+        this.operations = [];
+    }
+
+    /**
+     * Gives the result of precedence operations.  
+     */
+    calculateResult() {
+        let temporaryResult = this.numbers[0];
+        let finalResult = 0;
+        for (let i = 0; i < this.operations.length; i++) {
+            if (this.operations[i] === 'x') {
+                temporaryResult *= this.numbers[i + 1];
+            } else if (this.operations[i] === '/') {
+                temporaryResult /= this.numbers[i + 1];
             } else {
-                temporaryResult += sign * this.numbers[i + 1];
+                let sign = this.operations[i] === '+' ? 1 : -1;
+                if (i !== this.operations.length - 1 && this.operations[i + 1] === '/' || this.operations[i + 1] === 'x') {
+                    finalResult += temporaryResult;
+                    temporaryResult = sign * this.numbers[i + 1];
+                } else {
+                    temporaryResult += sign * this.numbers[i + 1];
+                }
+            }
+            if (i === this.operations.length - 1) {
+                finalResult += temporaryResult;
             }
         }
-        if (i === this.operations.length - 1) {
-            finalResult += temporaryResult;
-        }
+        return finalResult;
     }
-    return finalResult;
-}
 
-/**
- * Allows calculation with precedence operations.  
- */
-Precedence.prototype.precedenceOn = function() {
-    this.turnOn = true;
-}
+    /**
+     * Allows calculation with precedence operations.  
+     */
+    precedenceOn() {
+        this.turnOn = true;
+    }
 
-/**
- * Adds operation. 
- * @param {String}  operation. 
- */
-Precedence.prototype.addOperation = function(operation) {
-    this.operations.push(operation);
-}
+    /**
+     * Adds a operation. 
+     * @param {String}  operation. 
+     */
+    addOperation(operation) {
+        this.operations.push(operation);
+    }
 
-/**
- * Adds a number.
- * @param {Number}  number. 
- */
-Precedence.prototype.addNumber = function(number) {
-    this.numbers.push(Number(number));
-};
+    /**
+     * Adds a number.
+     * @param {Number}  number. 
+     */
+     addNumber(number) {
+        this.numbers.push(Number(number));
```

Comment:
As I wrote earlier, you can cast / parse number only when you are reading them from HTML elements. Rest of the code needs to work with numbers. So here you need to do `this.numbers.push(number)`.

If for any case `number` is a string here, then the problem is in the code which is invoking this method, not here.